tools/scan-images.sh

```diff
@@ -0,0 +1,72 @@
+set -eo pipefail
+
+# Check correct usage
+if [[ ! $2 ]]; then
+    echo "Usage: overcloud-ubuntu-upgrade.sh <os-distribution> <image-tag>"
+    exit 2
+fi
+
+set -u
+
+# Check that trivy is installed
+if ! trivy --version; then
+    echo 'Please install trivy: curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.49.1'
+fi
+
+# Make a fresh output directory
+mkdir -p image-scan-output
+
+# Get built container images
+docker image ls --filter "reference=ark.stackhpc.com/stackhpc-dev/$1-*:$2" > $1-scanned-container-images.txt
+
+# Make a file of imagename:tag
+images=$(grep --invert-match --no-filename ^REPOSITORY $1-scanned-container-images.txt | sed 's/ \+/:/g' | cut -f 1,2 -d:)
+
```

Comment:
The workflow currently assumes both image lists exist, better touch the files:
```
touch image-scan-output/clean-images.txt image-scan-output/dirty-images.txt
```

Reply:
Done